### PR TITLE
fix(curriculum): Added test and corrected solution for Restrict Possible Usernames challenge

### LINKED ISSUE #35842: Regular Expressions: Restrict Possible Usernames -- instructions wording doesn't make sense
**Describe your problem and how to reproduce it:**
The challenge says "Usernames have to be at least two characters long. A two-**letter** username can only use alphabet letter characters."

However, this should be changed to "Usernames have to be at least two characters long. A two-**character** username can only use alphabet letter characters."

Letter already means an alphabet letter, whereas character means alphanumeric.  
**Add a Link to the page with the problem:**
[See challenge](https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/regular-expressions/restrict-possible-usernames/)

**Tell us about your browser and operating system:**
* Browser Name: 
* Browser Version: 
* Operating System: 

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.
 
Closes #35842

In this PR the following changes were made:
* Added a new restriction to the challenge that valid characters are alphanumeric instead of **any** character.
* Changed verbiage in one of the last restriction from "two-letter" to "two-character".
* Added a new test to test for the situation where the first character is a letter and the last two characters are numbers which should pass with the correct solution with the instructions.
* Change the solution to one which satisfies all the existing tests plus the new one added above.

### CODE REVIEW

**Comment by dylanhamada** [comment]
Referencing the following regular expression:

`/^[a-z]([a-z]|[0-9]{2,}|[a-z]+\d*)$/i`

The first sequence in the capture group (`[a-z]`) can be omitted because it's redundant.

Then the regular expression is:

`/^[a-z]([0-9]{2,}|[a-z]+\d*)$/i`

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/restrict-possible-usernames.english.md [hunk 1]
```diff
@@ -67,7 +70,9 @@ let result = userCheck.test(username);
 <section id='solution'>
 
 ```js
-const userCheck = /^[A-Za-z]{2,}\d*$/;
+let username = "JackOfAllTrades";
+const userCheck = /^[a-z]([a-z]|[0-9]{2,}|[a-z]+\d*)$/i;
```
Comment (dylanhamada) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/restrict-possible-usernames.english.md]:
```suggestion
const userCheck = /^[a-z]([0-9]{2,}|[a-z]+\d*)$/i;
```

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Review by moT01 — APPROVED** [review]
Looks good to me @RandellDawson!

**Pull request merged**